### PR TITLE
Add com.corsixth.corsixth

### PR DESCRIPTION
**Pull request by rmnvgr**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
    - Read-only home directory access so that users can import the game assets
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link: https://github.com/CorsixTH/CorsixTH/issues/1416#issuecomment-835540527**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
    - Lua patch: They don't want to allow building shared library from their Makefile (http://lua-users.org/lists/lua-l/2006-03/msg00088.html)
    - CorsixTH patches: The app ID changed after the latest release

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

cc @TheCycoONE

### CODE REVIEW

**Comment by rmnvgr** [comment]
bot, build com.corsixth.corsixth

File: corsixth-appdata.patch [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+rename from CorsixTH/com.corsixth.CorsixTH.appdata.xml
+rename to CorsixTH/com.corsixth.corsixth.appdata.xml
+index 68819ccd94..b252d5bc78 100644
+--- a/CorsixTH/com.corsixth.CorsixTH.appdata.xml
++++ b/CorsixTH/com.corsixth.corsixth.appdata.xml
+@@ -1,6 +1,6 @@
+ <?xml version="1.0" encoding="UTF-8"?>
+ <component type="desktop">
+-  <id>com.corsixth.CorsixTH</id>
++  <id>com.corsixth.corsixth</id>
```
Comment (hfiguiere) [inline corsixth-appdata.patch:22]: So why not using this ID instead to not even have to rename?

File: corsixth-appdata.patch [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+   <!-- Copyright 2019 Artem Polishchuk <ego.cordatus@gmail.com> -->
+   <metadata_license>CC0</metadata_license>
+   <project_license>BSD-3-Clause AND MIT</project_license>
+@@ -24,14 +24,14 @@
+     <binary>corsix-th</binary>
+   </provides>
+   <releases>
+-    <release version="0.63" date="2019-05-24"></release>
++    <release version="0.64" date="2020-06-16"></release>
```
Comment (hfiguiere) [inline corsixth-appdata.patch:31]: Did you submit these upstream?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: corsixth-appdata.patch (hunk 1 above)
Comment (rmnvgr) [reply corsixth-appdata.patch:22]: Upstream changed their ID from `com.corsixth.CorsixTH` to `com.corsixth.corsixth`, but it happened after their latest release: https://github.com/CorsixTH/CorsixTH/issues/1416#issuecomment-835636263

I rename for this release, but it won't be necessary for future releases.

File: corsixth-appdata.patch (hunk 2 above)
Comment (rmnvgr) [reply corsixth-appdata.patch:31]: The two CorsixTH patches are actually upstream commits that I cherry-picked:
- https://github.com/CorsixTH/CorsixTH/commit/baec9533810026e55554ce7a6dc8e1817173ca88
- https://github.com/CorsixTH/CorsixTH/commit/8962dfc7880d08ee77c2f81dacdb30c9114f2204

They will be dropped with the next upstream release.

File: corsixth-appdata.patch (hunk 1 above)
Comment (hfiguiere) [reply corsixth-appdata.patch:22]: ok. make sense.

File: corsixth-appdata.patch (hunk 2 above)
Comment (hfiguiere) [reply corsixth-appdata.patch:31]: Excellent.

Comment (TheCycoONE) [reply corsixth-appdata.patch:31]: Did you also need https://github.com/CorsixTH/CorsixTH/commit/b595595439726f35b36004cfd6f41b3f61a3a009

File: corsixth-appdata.patch (hunk 1 above)
Comment (bilelmoussaoui) [reply corsixth-appdata.patch:22]: You should add the old id to `<provides><id>oldId</id></provides>`  so that the stores don't show the application twice

Comment (rmnvgr) [reply corsixth-appdata.patch:22]: Done, the patch has been submitted upstream here: https://github.com/CorsixTH/CorsixTH/pull/1871

**Comment by rmnvgr** [comment]
> I would consider adding innoextract (example: [flathub/io.openrct2.OpenRCT2@5ef5a99](https://github.com/flathub/io.openrct2.OpenRCT2/commit/5ef5a9924ca90a5193d43c6b81b81599759adcdd)) just so it's easier to extract the installer.

OpenRCT2 uses `innoextract` itself, I don't think that's the case with CorsixTH (@TheCycoONE?), users have to run it themselves. Is there a point of including it given that users may not know that it's included in the flatpak?

**Comment by barthalion** [comment]
Up to you; I ended up installing OpenRCT2 just to extract the installer here for testing. Probably doesn't make much sense if it's too opaque. Can be handled later anyway.

/merge @TheCycoONE

**Pull request closed without merging**